### DISCUSSION TITLE
Faces extraction from pyvista mesh

### DISCUSSION THREAD

**Discussion by sunitasaha** [pyvista/pyvista#4975] [Q&A]
```python
points = 3d points in numpy array
cloud = pv.PolyData(points)
sub_mesh = cloud.delaunay_3d().extract_geometry()
sub_mesh.plot(show_edges=True)
```
Output as expected:
![image](https://github.com/pyvista/pyvista/assets/70508935/30253bb9-a137-4544-a03c-60c0218ccab8)

Then if I want to assess the face information and try to test
```python
surface_faces = sub_mesh.faces
test_mesh = pv.PolyData(points,surface_faces)
test_mesh.plot(show_edges=True)
```

The results are messed up:
![image](https://github.com/pyvista/pyvista/assets/70508935/f5ea8ed4-8a60-4e5a-9c78-984ace1b0405)

Why the faces are not in place? However it does work for simple meshes with few points.

**Comment by banesullivan** [comment]
I suspect `points` is not in the same as `sub_mesh.points`. `sub_mesh.faces` is referring to `sub_mesh.points` not `points`. The delaunay+extract operations are altering the order of the points

**Reply by banesullivan** [reply]
`extract_geometry()` is geometry reducing, so you are definitely loosing points here from the 3D delaunay
